tinygrad/renderer/cstyle.py

```diff
@@ -42,7 +42,7 @@ class CStyleLanguage(NamedTuple):
   }
 
   # returns a str expression of the casted xs with the given type
-  def render_cast(self, x:List[str], var_dtype:DType) -> str:
+  def render_cast(self, x:List[str], var_dtype:DType, buf_dtype:Union[DType, None]=None) -> str:
```

Comment:
What is buf_dtype and why do we need it?

Reply:
it's supposed to be the original dtype of the value that's being casted. half2float is a special fn can only cast between those two dtypes, so we end up not only needing the input but also the output to check

Reply:
but the function body is still a wip.

in an ast that casts half2->float2 the args are `var_dtype=float2, buf_dtype=half`, buf_dtype should be a half2 instead, i'm looking to see if something like float2 could be implemented here too